cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -130,6 +143,39 @@ def shortest_path(self, lq1: 'cirq.Qid', lq2: 'cirq.Qid') -> Sequence['cirq.Qid'
         physical_shortest_path = self._physical_shortest_path(self._map[lq1], self._map[lq2])
         return [self._inverse_map[pq] for pq in physical_shortest_path]
 
-    @cached_method
+    @_compat.cached_method
     def _physical_shortest_path(self, pq1: 'cirq.Qid', pq2: 'cirq.Qid') -> Sequence['cirq.Qid']:
         return nx.shortest_path(self._induced_subgraph, pq1, pq2)
+
+    def _value_equality_values_(self):
+        graph_equality = (
+            tuple(self.device_graph.nodes),
+            tuple(self.device_graph.edges),
+            nx.is_directed(self.device_graph),
+        )
+        map_equality = tuple(sorted(self._map.items()))
+        return (graph_equality, map_equality)
+
+    def __str__(self) -> str:
+        return self.__repr__()
```

Comment:
nit: this is the default, so you can just define __repr__ and omit the definition of __str__